Lexplorer/Pages/NFTCollection.razor

```diff
@@ -0,0 +1,174 @@
+                           <MudCardActions>
+                               <MudButton Style="position:absolute;bottom:0;" Variant="Variant.Text" Color="Color.Primary" Link=@LinkHelper.GetObjectLinkAddress(slot)?.Item1>Go to NFT</MudButton>
+                           </MudCardActions>
+                    </MudCard>
+                </MudItem>
+            }
+        }
+        else
+        {
+            <MudText class="pa-3">No NFTs</MudText>
+        }
+        <MudHidden Breakpoint="Breakpoint.MdAndDown" Invert="true">
+            <MudPaper Width="100%" Class="mx-4">
+                <MudToolBar Class="justify-center">
+                    <MudPagination Size="Size.Small" ShowFirstButton="true" 
+                    BoundaryCount="1" 
+                    MiddleCount="1" 
+                    ShowLastButton="true" 
+                    Selected="@goToPage"
+                    Count="@goToNextPage"
+                    SelectedChanged="@((int page) => goToPage = page)"/>
+                 </MudToolBar>
+           </MudPaper>
+       </MudHidden>
+       <MudDivider DividerType="DividerType.Middle" Class="my-6"/>
+    </MudGrid>
+</MudContainer>
+
+@code {
+    [Parameter]
+    public string? tokenAddress { get; set; }
+
+    [Parameter]
+    [SupplyParameterFromQuery]
+    public string pageNumber { get; set; } = "1";
+
+    public int goToPage
+    {
+        get
+        {
+            return int.TryParse(pageNumber, out int np) ? np : 1;
+        }
+        set
+        {
+            pageNumber = value.ToString(); //set immediately for MudPagination to not switch forth and back
+            string URL = $"/collections/{tokenAddress}?pageNumber={value}";
+            NavigationManager.NavigateTo(URL);
+        }
+    }
+
+    public int goToNextPage
+    {
+        get
+        {
+            return int.TryParse(pageNumber, out int np) ? np + 1 : 1;
+        }
+    }
+
+    public readonly int pageSize = 12; //6 per row
+    private CancellationTokenSource? cts;
+    private IList<NonFungibleToken>? collectionNFTSlots { get; set; } = new List<NonFungibleToken>();
+    private Dictionary<string, NftMetadata> NFTdata { get; set; } = new Dictionary<string, NftMetadata>();
+    protected override async Task OnParametersSetAsync()
+    {
+        //cancel any previous OnParametersSetAsync which might still be running
+        cts?.Cancel();
+
+        using (CancellationTokenSource localCTS = new CancellationTokenSource())
+        {
+            //give future calls a chance to cancel us; it is now safe to replace
+            //any previous value of cts, since we already cancelled it above
+            cts = localCTS;
+            try
+            {
+                if (String.IsNullOrEmpty(tokenAddress)) return;
+                if (string.IsNullOrEmpty(pageNumber))
+                {
+                    pageNumber = "1";
+                }
+
+                string collectionNftKey = $"collection-{tokenAddress}-page-{pageNumber}";
```

Comment:
if you use gotoPage here instead of pageNumber, the above null check can be omitted. That's just a leftover on the account details page where this code came from ;-)

Reply:
hmm it doesnt seem to add the other pages if i get rid of the null check and use goToPage directly in there

Reply:
Can you explain? Because it surely works for me here.

Reply:
as per dicord. fixed in [abe7fd8](https://github.com/fudgebucket27/Lexplorer/pull/182/commits/abe7fd842eaf4fe21deb20f7974e35fda12bada2)